src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
+   auto error = nn_errno ();
+   if (error == ENOMEM) {
+      throw std::runtime_error("not enough memory to allocate NanoMsg");
```

Comment:
I think the library should handle exceptions locally by generally logging a warning. We can discuss.

In any case, you are using this class in places without a try-catch block.